src/utxocommit.cpp

```diff
@@ -0,0 +1,96 @@
+// Copyright (c) 2017 The Bitcoin developers
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+#include "utxocommit.h"
+
+#include "coins.h"
+#include "util.h"
+
+namespace {
+// Lazy, thread-safe singleton that wraps the secp256k1_context used for the
+// multisets
+class CSecp256k1Context {
+private:
+    secp256k1_context *context;
+
+    CSecp256k1Context() {
+        context = secp256k1_context_create(SECP256K1_CONTEXT_NONE);
+    }
+
+    ~CSecp256k1Context() { secp256k1_context_destroy(context); }
+
+public:
+    static secp256k1_context *GetContext() {
+        static CSecp256k1Context instance;
+        return instance.context;
+    }
+    CSecp256k1Context(CSecp256k1Context const &) = delete;
+    void operator=(CSecp256k1Context const &) = delete;
+};
+}
+
+// Constructs empty CUtxoCommit
+CUtxoCommit::CUtxoCommit() {
+    secp256k1_multiset_init(CSecp256k1Context::GetContext(), &multiset);
+}
+
+// Adds a TXO to multiset
+void CUtxoCommit::Add(const COutPoint &out, const Coin &element) {
+    CDataStream txo(SER_NETWORK, PROTOCOL_VERSION);
+    txo << out << element;
+    secp256k1_multiset_add(CSecp256k1Context::GetContext(), &multiset,
+                           (const uint8_t *)&txo[0], txo.size());
```

Comment:
can this call `txo.data()` instead of accessing `txo[0]`?

Reply:
Fixed